### ISSUE TITLE
Fusion EXRIO documentation

### ISSUE THREAD

**Issue by timmythetiny** [Psyop/Cryptomatte#66]
Where can I find documentation on the EXRIO pointer that's created in fusion/cryptomatte_utilities.lua?  There's nothing to be found in the scripting docs, user manual, or tool reference, and nor does Google find anything relevant.  Do I need access to the Fusion SDK to develop with EXRIO?

**Comment by AndrewHazelden** [comment]
@timmythetiny EXRIO is a new feature added to Fusion (Free) and Fusion Studio v9. It allows an uncompiled fuse module to read data directly from an EXR format image without needing to use a Loader node or Channel Booleans node to pre-process the data beforehand.

To the best of my knowledge, the Fusion scripting guide and SDK documentation notes have not been updated yet to cover the new API additions in Fusion 9. If you look on your hard disk you will see the "Fusion 8 Scripting Manual.pdf" is still included with Fusion 9.

The only other publicly viewable information on EXRIO besides the code found in Cryptomatte is the `External Matte Saver.fuser` example that is located on your hard disk at:

`C:/Program Files/Blackmagic Design/Fusion 9/Fuses/Blackmagic/External Matte Saver.fuser`

**Comment by timmythetiny** [comment]
@AndrewHazelden Thank you.  I assume the contributors on this decoder have access to the latest API docs from Blackmagic under NDA?

**Comment by jonahfriedman** [comment]
@timmythetiny  I have a locked briefcase here which I can't comment on the contents of, because the burly man it's handcuffed to is reading this over my shoulder. 

(Seriously, they worked with Cedric to get this working, and I suspect they just haven't gotten around to writing formal docs on this yet).

**Comment by jonahfriedman** [comment]
Anyway, I'm going to go ahead and close this since it's not a Cryptomatte issue, but feel free to keep discussing it.

**Comment by cedricduriau** [comment]
Hi @timmythetiny, I was lucky enough to go back and forth with the developer(s) of Fusion about this topic. EXRIO was a bit of a request/lucky gift from them to let Cryptomatte and Fusion support a multi channel workflow.

I don't have access to the SDK myself, this is true, I did my absolute best to reverse engineer the module from the fuse you mention and with a lot of hours into it and some tips and tricks from the developers, I was able to implement this into Cryptomatte.

If you have any questions, I will be happy to help you out.
If you wish to discuss this by mail: duriau.cedric@live.be

Cheers
Cedric
